From a1a9587b5e403c03d4d22b09d05b190b4bf6e0a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20S=C5=82awi=C5=84ski?=
 <amadeuszx.slawinski@intel.com>
Date: Fri, 23 Nov 2018 13:55:45 +0100
Subject: [PATCH 232/292] ASoC: core: Tell codec that jack is being removed
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When component is being removed we should disable jack, otherwise some
codecs will try to trigger interrupt using freed structures.

Change-Id: Ia268f7799361d9c6c594f254649e7a7793445e40
Signed-off-by: Amadeusz Sławiński <amadeuszx.slawinski@intel.com>
Reviewed-on:
---
 sound/soc/soc-core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 3814f9359794..bc8665f5caf7 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -942,6 +942,8 @@ static void soc_remove_component(struct snd_soc_component *component)
 
 	list_del(&component->card_list);
 
+	snd_soc_component_set_jack(component, NULL, NULL);
+
 	if (component->driver->remove)
 		component->driver->remove(component);
 
-- 
2.21.0

